src/ncp/hdlc.hpp

```diff
@@ -54,48 +54,97 @@ namespace Hdlc {
 class Encoder
 {
 public:
+
     /**
-     * This method begins an HDLC frame and puts the initial bytes into @p aOutBuf.
+     * This class defines a write iterator into a buffer used by Encoder.
      *
-     * @param[in]     aOutBuf     A pointer to the output buffer.
-     * @param[inout]  aOutLength  On entry, the output buffer size; On exit, the output length.
+     * Hdlc users should sub-class this to add the actual buffer space.
+     */
+    class BufferWriteIterator
+    {
+    public:
+
+        /**
+         * This method writes a byte to the buffer and updates the iterator (if space is available).
+         *
+         * @retval kThreadError_None    Successfully wrote the byte and updates the iterator.
+         * @retval kThreadError_NoBufs  Insufficient buffer space.
+         *
+         */
+        ThreadError WriteByte(uint8_t aByte);
+
+        /**
+         * This method checks if there is buffer space available to write @p aWriteLength bytes.
+         *
+         * param[in] aWriteLength       Number of bytes to write.
+         *
+         * @retval true                 Enough buffer space available to write the requested number of bytes.
+         * @retval false                Insufficient buffer space to write the requested number of bytes.
+         *
+         */
+        bool CanWrite(uint16_t aWriteLength);
```

Comment:
Make this a `const`?  For example: `bool CanWrite(uint16_t aWriteLength) const;`